zarr/util.py

```diff
@@ -243,6 +243,16 @@ def normalize_order(order: str) -> str:
     return order
 
 
+def normalize_dimension_separator(sep: Optional[str]) -> Optional[str]:
+    if sep is None:
+        return None
+    elif sep not in (".", "/"):
+        raise ValueError(
+            "dimension_separator must be either '.' or '/', found: %r" % sep)
+    else:
+        return sep
```

Comment:
The  `if sep is None` can be folded into the else.

Not found of the "normalize", normalize confer to me that it might change it (like / to \ on windows); 
maybe just a check ? 

```
assert sep in (".", "/", None)
``` 
might seem enough as it is only used in 2 places ?

Reply:
And once we have a base store we can turned it into a validated property.

Reply:
The original commit turned `None` into `.` so it was much more normalize-y. I'm not completely sure that logic won't come back though. (And this way I could write a unit test for it!)